examples/with-vite-plugin-pwa/astro.config.mjs

```diff
@@ -1,9 +1,79 @@
 import { defineConfig } from 'astro/config';
-import { VitePWA } from 'vite-plugin-pwa';
+import pwa from '@astrojs/pwa';
+import replace from '@rollup/plugin-replace';
+
+const pwaOptions = {
+	mode: 'development',
+	base: '/',
+	registerType: 'autoUpdate',
+	includeAssets: ['favicon.svg'],
+	manifest: {
+		name: 'Astro PWA',
+		short_name: 'Astro PWA',
+		description: 'Astro PWA Auto Generate',
+		theme_color: '#ffffff',
+		icons: [
+			{
+				src: 'pwa-192x192.png', // <== don't add slash, for testing
+				sizes: '192x192',
+				type: 'image/png',
+			},
+			{
+				src: '/pwa-512x512.png', // <== don't remove slash, for testing
+				sizes: '512x512',
+				type: 'image/png',
+			},
+			{
+				src: 'pwa-512x512.png', // <== don't add slash, for testing
+				sizes: '512x512',
+				type: 'image/png',
+				purpose: 'any maskable',
+			},
+		],
+	},
+	devOptions: {
+		enabled: process.env.SW_DEV === 'true',
+		/* when using generateSW the PWA plugin will switch to classic */
+		type: 'module',
+		navigateFallback: '/',
+	},
+}
+
+const replaceOptions = {
+	__DATE__: `${new Date().toISOString()}`,
+	include: [/\.(ts|astro)$/],
+}
+
+const reload = process.env.RELOAD_SW === 'true'
+
+if (process.env.SW === 'true') {
+	pwaOptions.srcDir = 'src'
+	pwaOptions.filename = 'claims-sw.ts'
+	pwaOptions.strategies = 'injectManifest'
+	pwaOptions.manifest.description = 'Astro PWA Inject Manifest'
+} 
+
+if (reload) {
+	// @ts-ignore
+	replaceOptions.__RELOAD_SW__ = 'true'
+}
 
 // https://astro.build/config
 export default defineConfig({
+	integrations: [pwa(pwaOptions)],
 	vite: {
-		plugins: [VitePWA()],
-	},
+		resolve: {
+			alias: {
+				'workbox-window': './node_modules/workbox-window/build/workbox-window.prod.es5.mjs'
+			},
+		},
+		optimizeDeps: {
+			include: ['workbox-window']
+		},
```

Comment:
Will `workbox-window` always be required? If possible, I'd say this config logic should move to the `@astrojs/pwa` integration.

Reply:
The problem is Vite 2.9.9, I include it on optimizedDeps to avoid multiple reloads for new depedency discovery: we can fix it including the dep and/or upgrading to latest Vite 2.9 (IIRC included in 2.9.10: 2.9.14).

Reply:
Should be fixed by https://github.com/withastro/astro/pull/3861 in that case